apps/dotcom/src/utils/defineLoader.tsx

```diff
@@ -1,8 +1,8 @@
 import { LoaderFunctionArgs, useLoaderData } from 'react-router-dom'
 
 export function defineLoader<T>(_loader: (args: LoaderFunctionArgs) => Promise<T>): {
-	loader(args: LoaderFunctionArgs): Promise<{ [specialSymbol]: T }>
-	useData(): Exclude<T, Response>
+	loader(this: void, args: LoaderFunctionArgs): Promise<{ [specialSymbol]: T }>
```

Comment:
Not sure I really like this, but it seems the way to handle these cases. We could also disable them on the usage side of things, but that's also not great.